File: KAKAO BLIND RECRUITMENT/[1차] 비밀지도.py
def solution(n, arr1, arr2):
    temp_1, temp_2 = [], []
    
    for arr in arr1:
        ls = [] # arr1의 원소 하나 당 이진변환 결과를 담을 리스트
        tmp = bin(arr)[2:] # 이진변환한 결과
        if len(tmp) < n: # 만약 자릿수가 주어진 n보다 모자를 경우, 앞 자리에 모자란 자릿수만큼 0 채워주기
            tmp = '0'*(n-len(tmp)) + tmp
        for t in tmp:
            ls.append(t)
        temp_1.append(ls)    
            
    for arr in arr2:
        ls = [] 
        tmp = bin(arr)[2:]
        if len(tmp) < n:
            tmp = '0'*(n-len(tmp)) + tmp
        for t in tmp:
            ls.append(t)
        temp_2.append(ls)              
    
    answer = []
    for i in range(len(temp_1)):
        sum_ls = '' # 두 개의 결과를 합친 것
        for j in range(len(temp_1[i])):
            if temp_1[i][j] == '0' and temp_2[i][j] == '0':
                sum_ls += ' '
            elif temp_1[i][j] == '1' or temp_2[i][j] == '1':
                sum_ls += '#'       
        answer.append(sum_ls)
            
    return answer

File: KAKAO BLIND RECRUITMENT/[1차] 비밀지도.java
// https://school.programmers.co.kr/learn/courses/30/lessons/17681
import java.util.*;
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        String[] array1 = convertArr(arr1, n);
        String[] array2 = convertArr(arr2, n);
      
        for(int i=0; i<n; i++) {
            String str = "";
            for(int j=0; j<n; j++) {
                if (array1[i].charAt(j) == '1' || array2[i].charAt(j) == '1') {
                    str += "#";
                } else if (array1[i].charAt(j) == '0' && array2[i].charAt(j) == '0') {
                    str += " ";
                }
            }
            answer[i] = str;
        }
        
        return answer;
    }
    
    public String[] convertArr(int[] arr, int n) {
        
        String[] array = new String[n];
        
        for(int i=0; i<arr.length; i++) {
            String str = Integer.toBinaryString(arr[i]);
            String zero = "";
            if (str.length() < n) {
                for(int j=0; j<n-str.length(); j++) 
                    zero += "0";
            }
            str = zero + str;
            array[i] = str;
        }
        
        return array;
    }
}
